File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
       int maxSum = nums[0];
       int currSum = nums[0];

       for(int i=1; i<nums.length; i++){
            int val = nums[i];
            currSum = Math.max(val, currSum+val);
            maxSum = Math.max(maxSum, currSum);
       }

       return maxSum; 

       
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        res = max(nums)
        cur_max, cur_min = 0, 0
        for num in nums:
            tmp = cur_max + num
            cur_max = max(cur_max + num, cur_min + num, num)
            cur_min = min(cur_min + num, tmp, num)
            res = max(cur_max, res)
        return res